vendor/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/machinehealthcheck/machinedisruptionbudget.go

```diff
@@ -0,0 +1,149 @@
+		// that will give use possibility to make this test work
+		if isKubemarkProvider {
+			glog.V(2).Info("Can not run this tests with the 'KubeMark' provider")
+			Skip("Can not run this tests with the 'KubeMark' provider")
+		}
+
+		err = e2e.CreateOrUpdateTechPreviewFeatureGate()
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Getting worker node")
+		workerNodes, err := e2e.GetWorkerNodes(client)
+		Expect(err).ToNot(HaveOccurred())
+
+		readyWorkerNodes := e2e.FilterReadyNodes(workerNodes)
+		Expect(readyWorkerNodes).ToNot(BeEmpty())
+
+		workerNode = &readyWorkerNodes[0]
+		glog.V(2).Infof("Worker node %s", workerNode.Name)
+
+		By("Getting worker machine")
+		workerMachine, err := e2e.GetMachineFromNode(client, workerNode)
+		Expect(err).ToNot(HaveOccurred())
+		glog.V(2).Infof("Worker machine %s", workerMachine.Name)
+
+		By("Geting worker machine set")
+		workerMachineSet, err = e2e.GetMachineMachinesSet(workerMachine)
+		Expect(err).ToNot(HaveOccurred())
+
+		glog.V(2).Infof("Create machine health check with label selector: %s", workerMachine.Labels)
+		err = e2e.CreateMachineHealthCheck(workerMachine.Labels)
+		Expect(err).ToNot(HaveOccurred())
+
+		unhealthyConditions := &conditions.UnhealthyConditions{
+			Items: []conditions.UnhealthyCondition{
+				{
+					Name:    "Ready",
+					Status:  "Unknown",
+					Timeout: "60s",
+				},
+			},
+		}
+		glog.V(2).Infof("Create node-unhealthy-conditions configmap")
+		err = e2e.CreateUnhealthyConditionsConfigMap(unhealthyConditions)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("updates MDB status", func() {
+		minAvailable := int32(3)
+		testMdb = e2e.NewMachineDisruptionBudget(
+			mdbName,
+			workerMachineSet.Spec.Selector.MatchLabels,
+			&minAvailable,
+			nil,
+		)
+		By("Creating MachineDisruptionBudget")
+		err := client.Create(context.TODO(), testMdb)
+		Expect(err).ToNot(HaveOccurred())
+
+		updateMdb := &healthcheckingv1alpha1.MachineDisruptionBudget{}
+		Eventually(func() int32 {
+			key := types.NamespacedName{
+				Name:      mdbName,
+				Namespace: e2e.TestContext.MachineApiNamespace,
+			}
+			err := client.Get(context.TODO(), key, updateMdb)
+			if err != nil {
+				return 0
+			}
+			return updateMdb.Status.ExpectedMachines
+		}, 120*time.Second, time.Second).Should(Equal(*workerMachineSet.Spec.Replicas))
+
+		currentHealthy := updateMdb.Status.CurrentHealthy
+		Expect(currentHealthy).To(Equal(workerMachineSet.Status.ReadyReplicas))
+
+		By(fmt.Sprintf("Stopping kubelet service on the node %s", workerNode.Name))
+		err = e2e.StopKubelet(workerNode.Name)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Waiting until worker machine will have unhealthy node
+		Eventually(getCurrentHealthyMachines, 6*time.Minute, 10*time.Second).Should(Equal(currentHealthy - 1))
+
+		// Waiting until machine set will create new healthy machine
+		Eventually(getCurrentHealthyMachines, 15*time.Minute, 30*time.Second).Should(Equal(currentHealthy))
```

Comment:
why 15? something lower like 5 would be reasonable, even if it pass with 15 that's not a minimum acceptable bar

Reply:
at least I saw that it is a reasonable timeout to create, provision and append to the cluster the new machine.